### PR TITLE
fix: address review findings for CommandRegistry parse safety

### PR DESCRIPTION
**Pull request by currentsuspect**
## Summary

Addresses all CodeRabbit/Copilot review findings on the original CommandRegistry parse safety PR (#214, already merged).

### 1. Missing flag crash path (critical)
- All command factories now use `requireFlag()` before parsing
- Previously `flags.at()` was called BEFORE the safe parser, so missing keys still threw `std::out_of_range`

### 2. `safeStof` accepts NaN/Inf (critical)
- Now rejects `nan`, `NaN`, `inf`, `Inf`, `-inf`, `infinity`
- Previously these would pass through and poison audio state

### 3. `safeStoull` accepts negative strings
- Now rejects strings starting with `-` (e.g. `"-1"` wraps to ULLONG_MAX)

### 4. Dead code removal
- Removed unused `getFlag()` helper

### 5. Regression tests
- Added `CommandRegistryParseSafetyTest.cpp` covering all parse helpers

## Testing performed
- All CI checks pass (Linux, Windows, macOS, sanitizers, clang-tidy)
- New test executable validates all parse helper edge cases

## Docs updated?
No

## Risk/rollback notes
- Low risk: purely defensive changes
- Rollback: revert commit if any command parsing regressions found

## CommandRegistry parse safety improvements

**Key changes:**

- **Added safe parsing helpers** (`requireFlag()`, `safeStoi()`, `safeStof()`, `safeStoull()`) in CommandRegistry.cpp to prevent crashes from missing flags and malformed input
  - `requireFlag()` safely returns `std::nullopt` for missing keys (prevents `std::out_of_range` from `.at()` calls)
  - `safeStof()` explicitly rejects non-finite floats (NaN, Inf, -Inf, infinity)
  - `safeStoull()` rejects negative strings to prevent unsigned integer wrapping

- **Updated all command factories** (15+ factories) to call `requireFlag()` before parsing, catching missing required flags early and returning nullptr on any parse failure

- **Removed dead code**: unused `getFlag()` helper

- **Added comprehensive regression test suite** in `Tests/Commands/CommandRegistryParseSafetyTest.cpp` covering:
  - Valid integer/float/unsigned inputs
  - Malformed inputs (empty strings, letters, trailing whitespace, non-numeric chars)
  - NaN/Inf rejection for float parser
  - Negative string rejection for unsigned parser
  - `requireFlag()` nullopt behavior for missing keys

**Risk assessment:** Low—defensive changes with full test coverage and passing CI (Linux, Windows, macOS, sanitizers, clang-tidy)

[Review Change Stack](https://app.coderabbit.ai/change-stack/currentsuspect/Aestra/pull/222?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

**Review by currentsuspect — COMMENTED** [review]
@coderabbitai any blocker? if not approve please

File: AestraAudio/src/Commands/CommandRegistry.cpp [hunk 1]
```diff
@@ -113,109 +164,176 @@ void CommandRegistry::initialize(TrackManager* trackManager) {
...
-        bool state = parseFlagBool(flags.at("state"));
-        MixerChannel* ch = tm->getChannel(static_cast<size_t>(trackIndex));
+        auto trackRaw = requireFlag(flags, "track");
+        if (!trackRaw) return nullptr;
+        auto trackOpt = safeStoi(*trackRaw);
+        if (!trackOpt) return nullptr;
+        auto stateRaw = requireFlag(flags, "state");
+        if (!stateRaw) return nullptr;
+        bool state = parseFlagBool(std::string(*stateRaw));
+        MixerChannel* ch = tm->getChannel(static_cast<size_t>(*trackOpt));
```
[the remark this answers is not in the document]
Comment (currentsuspect) [reply AestraAudio/src/Commands/CommandRegistry.cpp:214]: @copilot apply changes based on [this feedback](https://github.com/currentsuspect/Aestra/pull/222#discussion_r3254691193)

File: AestraAudio/src/Commands/CommandRegistry.cpp [hunk 2]
```diff
@@ -33,6 +35,52 @@ bool parseFlagBool(const std::string& s) {
...
+}
+
+// Safe parsing helpers that return std::nullopt on malformed input
+std::optional<int> safeStoi(std::string_view s) {
+    if (s.empty()) return std::nullopt;
+    try {
+        size_t pos = 0;
+        int val = std::stoi(std::string(s), &pos);
+        if (pos != s.size()) return std::nullopt;
+        return val;
```
[the remark this answers is not in the document]
Comment (currentsuspect) [reply AestraAudio/src/Commands/CommandRegistry.cpp:57]: @copilot apply changes based on [this feedback](https://github.com/currentsuspect/Aestra/pull/222#discussion_r3254691172)
